djpsa/sync/sync.py

```diff
@@ -73,6 +73,49 @@ def __init__(self):
         self.synced_ids = set()
 
 
+class SyncRelatedMixin:
+    """
+    Mixin for synchronizers that need to sync related objects.
+    """
+
+    def sync_related(self, instance, sync_classes=None):
+        """
+        Sync related objects for the given instance.
+        """
+        sync_classes = sync_classes + self.get_related_synchronizers(instance) \
```

Comment:
This is a verbose way to write this line. The `if` is buried on the second line. I would just do:

```
if not sync_classes:
    sync_classes = []
all_sync_classes = sync_classes + self.get_related ...
```

(Python devs should recognize the first 2 lines as necessary because it's not a good practice to use a list as default method args.)

Also I don't think it's good to re-use an argument name for another purpose, here adding a second list. (That's why I used `all_sync_classes` above`.)

But a more overarching question is- why does this need to get the list of sync classes from 2 places?

And the final big question- is this even used? The only occurrence of `sync_related` in the code is this definition. I see no usages.

Reply:
Yeah, this was a bit of YAGNI. I was adding that to make it easily extensible into the future if we wanted to override the behaviour without refactoring or code duplication, but it might not be needed here.